sdk/core/core/test/cmocka/test_az_aad.c

```diff
@@ -63,6 +67,19 @@ static void test_az_aad_request_token()
   // Calling az_aad_request_token(_az_http_request* ref_request, _az_token* out_token);
   _az_token t = { 0 };
   _az_http_request r = { 0 };
+
+  uint8_t headers[400];
+  uint8_t body[400];
+  assert_return_code(
+      az_http_request_init(
+          &r,
+          &az_context_app,
+          az_http_method_get(),
+          AZ_SPAN_FROM_STR("url"),
+          AZ_SPAN_FROM_BUFFER(headers),
+          AZ_SPAN_FROM_BUFFER(body)),
+      AZ_OK);
```

Comment:
Added this to make this tests pass
Since now we are adding an extra header, we need the request to have a valid buffer and I was using null for it because we were not adding anything